rp/src/cli.rs

```diff
@@ -0,0 +1,458 @@
+                        );
+                    }
+                }
+                _ => {
+                    return fatal(
+                        &format!("Unknown option {}", x),
+                        Some(CommandType::Exchange),
+                    )
+                }
+            }
+        }
+
+        Ok(peer)
+    }
+}
+
+impl ExchangeOptions {
+    pub fn parse(mut args: &mut Peekable<impl Iterator<Item = String>>) -> Result<Self, String> {
+        let mut options = ExchangeOptions::default();
+
+        if let Some(private_keys_dir) = args.next() {
+            options.private_keys_dir = PathBuf::from(private_keys_dir);
+        } else {
+            return fatal(
+                "Required positional argument: PRIVATE_KEYS_DIR",
+                Some(CommandType::Exchange),
+            );
+        }
+
+        while let Some(x) = args.next() {
+            let x = x.as_str();
+
+            match x {
+                "dev" => {
+                    if let Some(device) = args.next() {
+                        options.dev = Some(device);
+                    } else {
+                        return fatal("dev option requires parameter", Some(CommandType::Exchange));
+                    }
+                }
+                "listen" => {
+                    if let Some(addr) = args.next() {
+                        if let Ok(addr) = addr.parse::<SocketAddr>() {
+                            options.listen = Some(addr);
+                        } else {
+                            return fatal(
+                                "invalid parameter for listen option",
+                                Some(CommandType::Exchange),
+                            );
+                        }
+                    } else {
+                        return fatal(
+                            "listen option requires parameter",
+                            Some(CommandType::Exchange),
+                        );
+                    }
+                }
+                "peer" => {
+                    let peer = ExchangePeer::parse(&mut args)?;
+                    options.peers.push(peer);
+                }
+                _ => {
+                    return fatal(
+                        &format!("Unknown option {}", x),
+                        Some(CommandType::Exchange),
+                    )
+                }
+            }
+        }
+
+        Ok(options)
+    }
+}
+
+impl Cli {
+    pub fn parse(mut args: Peekable<impl Iterator<Item = String>>) -> Result<Self, String> {
+        let mut cli = Cli::default();
+
+        let _ = args.next(); // skip executable name
+
+        while let Some(x) = args.next() {
+            let x = x.as_str();
+
+            match x {
+                "verbose" => {
+                    cli.verbose = true;
+                }
+                "explain" => {
+                    eprintln!("WARN: the explain argument is no longer supported");
```

Comment:
I think it is fine if we remove the `explain`. Quines can be hard to achieve in compiled languages.

Reply:
Should the warning message stay or go?

Reply:
I think having the warning here is  fine.